cmd/kubeadm/app/features/features_test.go

```diff
@@ -129,7 +129,7 @@ func TestNewFeatureGate(t *testing.T) {
 func TestValidateVersion(t *testing.T) {
 	var someFeatures = FeatureList{
 		"feature1": {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Beta}},
-		"feature2": {FeatureSpec: featuregate.FeatureSpec{Default: true, PreRelease: featuregate.Alpha}, MinimumVersion: constants.MinimumControlPlaneVersion.WithPreRelease("alpha.1")},
+		"feature2": {FeatureSpec: featuregate.FeatureSpec{Default: true, PreRelease: featuregate.Alpha}, MinimumVersion: version.MustParseSemantic("v1.17.0").WithPreRelease("alpha.1")},
```

Comment:
i think this is fine, but wonder why it broke related to this change:

> Still I had to touch seemingly unrelated features_test.go to break the import cycle. Let me know if it's better to submit a separate PR for that.

Reply:
This `features` imports `constants` which imports `kubeadmapi` which is going to import `features` in this PR.

Reply:
ok, understood.

Reply:
Actually `GetDNSVersion` should be in the DNS addon package, but that way we get circular dependency through the images package...